contracts/fungible-tokens/andromeda-cw20-staking/src/contract.rs

```diff
@@ -273,6 +285,139 @@ fn execute_add_reward_token(
         .add_attribute("added_token", reward_token_string))
 }
 
+fn execute_remove_reward_token(
+    ctx: ExecuteContext,
+    reward_token_string: String,
+) -> Result<Response, ContractError> {
+    let ExecuteContext {
+        deps, info, env, ..
+    } = ctx;
+    let contract = ADOContract::default();
+    ensure!(
+        contract.is_owner_or_operator(deps.storage, info.sender.as_str())?,
+        ContractError::Unauthorized {}
+    );
+
+    // Set reward token as inactive
+    match REWARD_TOKENS.load(deps.storage, &reward_token_string) {
+        Ok(mut reward_token) if reward_token.is_active => {
+            // Need to save current status before setting reward token as inactive
+            // This is important in case the reward token is allocated token
+            let state = STATE.load(deps.storage)?;
+            update_global_index(
+                &deps.querier,
+                Milliseconds::from_seconds(env.block.time.seconds()),
+                env.contract.address,
+                &state,
+                &mut reward_token,
+            )?;
+
+            reward_token.is_active = false;
+            REWARD_TOKENS.save(deps.storage, &reward_token_string, &reward_token)?;
+        }
+        _ => {
+            return Err(ContractError::InvalidAsset {
+                asset: reward_token_string,
+            })
+        }
+    }
+
+    let mut config = CONFIG.load(deps.storage)?;
+    config.number_of_reward_tokens -= 1;
+    CONFIG.save(deps.storage, &config)?;
+
+    Ok(Response::new()
+        .add_attribute("action", "remove_reward_token")
+        .add_attribute("removed_token", reward_token_string))
```

Comment:
Would it be useful to add the new number of reward tokens as an attribute?

Reply:
resolved